core-ktx/src/main/java/com/amplifyframework/kotlin/Storage.kt

```diff
@@ -0,0 +1,118 @@
+import com.amplifyframework.storage.StorageException
+import com.amplifyframework.storage.options.StorageDownloadFileOptions
+import com.amplifyframework.storage.options.StorageGetUrlOptions
+import com.amplifyframework.storage.options.StorageListOptions
+import com.amplifyframework.storage.options.StorageRemoveOptions
+import com.amplifyframework.storage.options.StorageUploadFileOptions
+import com.amplifyframework.storage.options.StorageUploadInputStreamOptions
+import com.amplifyframework.storage.result.StorageDownloadFileResult
+import com.amplifyframework.storage.result.StorageGetUrlResult
+import com.amplifyframework.storage.result.StorageListResult
+import com.amplifyframework.storage.result.StorageRemoveResult
+import com.amplifyframework.storage.result.StorageTransferProgress
+import com.amplifyframework.storage.result.StorageUploadFileResult
+import com.amplifyframework.storage.result.StorageUploadInputStreamResult
+import java.io.File
+import java.io.InputStream
+import kotlinx.coroutines.ExperimentalCoroutinesApi
+import kotlinx.coroutines.FlowPreview
+import kotlinx.coroutines.flow.Flow
+import kotlinx.coroutines.flow.first
+import kotlinx.coroutines.flow.flattenMerge
+import kotlinx.coroutines.flow.flowOf
+import kotlinx.coroutines.flow.map
+import kotlinx.coroutines.flow.onEach
+
+@Suppress("unused")
+interface Storage {
+    @Throws(StorageException::class)
+    suspend fun getUrl(
+        key: String,
+        options: StorageGetUrlOptions = StorageGetUrlOptions.defaultInstance()
+    ):
+        StorageGetUrlResult
+
+    @ExperimentalCoroutinesApi
+    @FlowPreview
+    fun downloadFile(
+        key: String,
+        local: File,
+        options: StorageDownloadFileOptions = StorageDownloadFileOptions.defaultInstance()
+    ):
+        InProgressStorageOperation<StorageDownloadFileResult>
+
+    @ExperimentalCoroutinesApi
+    @FlowPreview
+    fun uploadFile(
+        key: String,
+        local: File,
+        options: StorageUploadFileOptions = StorageUploadFileOptions.defaultInstance()
+    ):
+        InProgressStorageOperation<StorageUploadFileResult>
+
+    @FlowPreview
+    @ExperimentalCoroutinesApi
+    fun uploadInputStream(
+        key: String,
+        local: InputStream,
+        options: StorageUploadInputStreamOptions = StorageUploadInputStreamOptions.defaultInstance()
+    ): InProgressStorageOperation<StorageUploadInputStreamResult>
+
+    @Throws(StorageException::class)
+    suspend fun remove(
+        key: String,
+        options: StorageRemoveOptions = StorageRemoveOptions.defaultInstance()
+    ): StorageRemoveResult
+
+    @Throws(StorageException::class)
+    suspend fun list(
+        path: String,
+        options: StorageListOptions = StorageListOptions.defaultInstance()
+    ): StorageListResult
+
+    @FlowPreview
+    data class InProgressStorageOperation<T>(
+        private val results: Flow<T>,
+        private val progress: Flow<StorageTransferProgress>,
```

Comment:
Make this public, so customers can access it?  I don't see a way for customers to access progress, currently.
```suggestion
       val progress: Flow<StorageTransferProgress>,
```

Reply:
Whoops! I actually have another commit, where I go back and update this mechanism. Yes, you're absolutely right. This current PR doesn't expose any ability for the customer to view progress. Will update.